manifests/charts/ztunnel/values.yaml

```diff
@@ -12,6 +12,10 @@ defaults:
   # If Image contains a "/", it will replace the entire `image` in the pod.
   image: ztunnel
 
+  # resourceName, if set, will override the naming of resources. If not set, will default to the release name.
+  # It is recommended to not set this; this is primarily for backwards compatibility.
+  resourceName: ""
```

Comment:
Why do we need this? In all our docs we are already explicitly setting the `Release.Name` to `ztunnel`?

e.g.`helm install ztunnel`

if back compat is important, just continue to install with the fixed name, we don't need another value for it.

Reply:
I was thinking: before, I did `helm install foo ..`. My ds is called ztunnel.

Now if I upgrade it will change the name?

Reply:
I don't think so? I think `helm upgrade ztunnel` expects there to _be_ an existing release named `ztunnel` - you cannot change the release name on upgrade.

Reply:
I am not talking about the `helm upgrade ztunnel` case -- I agree that is fine. But the case they picked a release name other than `ztunnel`

Reply:
> I was thinking: before, I did helm install foo ... My ds is called ztunnel.
> But the case they picked a release name other than ztunnel

Oh, I see. 

Still not sure how that matters, the new version will have resources with new names?

Reply:
yeah then would be duplicated/have a downtime during the rollout (not sure if helm prunes them?) was what I was thinking. Maybe its overkill

Reply:
If all the dependent resources are duplicated by-name I don't think it should matter - it would/should be a totally parallel set of resources, it's not really riskier than the typical in-place upgrade.

If Helm doesn't prune them just documenting the need for manual removal if you did something that our docs do not cover in the release notes is enough - it's hard to remove values once we add them and every value we add needs to be explained, so I would personally rather not.

Reply:
(if it does not prune the old resources, and just mentioning the workaround in the relnotes for the handful of people that have this problem isn't acceptable, I withdraw my objection)

Reply:
One other reason to possibly keep it is istioctl currently sets the release of everything to `istio`.

We could change it, but it impacts the labels..

Reply:
> One other reason to possibly keep it is istioctl currently sets the release of everything to istio.

Ok, then we really probably shouldn't use `Release.name` for the resource name at all.

In that case, a `revision-lite` suffix makes more sense.

`{{ define "ztunnel.release-name" }}ztunnel-{{ .Values.customName| default "" }}{{ end }}

Reply:
I think the requirements I want to meet are:
1. existing users must get "ztunnel" as the name for back compat
2. new users probably also should since it's the name we like
3. users should be able to install multiple ztunnels 

so that seems to suggest we either take the approach of this PR, or ignore release.name but allow overriding the "ztunnel" constant?

Reply:
> existing users must get "ztunnel" as the name for back compat
> new users probably also should since it's the name we like
> users should be able to install multiple ztunnels

Yes to all of those, but even in the case of No. 3 I'd like to retain a fixed name prefix, since we don't support fully  customizable component names for other stuff AFAIK (though maybe we should), and this may be a bit confusing for people.

I'm not too fussed about it tho.